---
title: "Analysis of MK2 transcript levels in TCGA-LUAD (NSCLC) dataset"
author: "Karthik Suresh"
date: "November 18, 2021 - Revision 4"
output:
  pdf_document:
    toc: yes
  html_document:
    theme: united
    toc: yes
---

Revisions (since rev2):

1. We've added the concept of validating RNASeq counts using a secondary extraction process. Using TCGAbiolinks, we extracted normalized RSEM reads and compared these reads to those obtained via OncoLnc to ensure that the read counts were largely similar at a patient-to-patient level

2. The original scatter plot showing differences in MK2 is restricted to only "Adenocarcinomas" within LUAD. We've applied that restriction to all the regression analyses as well

3. Some small clean-ups to the code

4. We've re-analyzed the proportion dead by tertiles of MK2 (instead of quantiles)

5. One advantage of downloading TCGA data is that we also happened to grab MK2 transcript levels from 50 samples with normal tissue controls, thus allowing us to calculate fold change in MK2 transcript levels between normal and tumor in matched samples. 

6. R censoring is now performed at one year and two year. This is in part to harmonize analyses with the validation cohort, where censoring could only be done at 2 years. 


# Data retrieval and cleaning

First, we are downloading clinical data from TCGA-LUAD dataset using the TCGAbiolinks API. For the MK2 transcript data, I grabbed these data from oncoLnc.org using "MAPKAPK2" as the gene name and TCGA-LUAD as the cohort. 

THe code used to generate the TCGA manual reads is provided elsewhere. 


## Retrieve data from TCGA dataset, _join it with the MK2 transcript data, with additional joins for the manually extracted reads from TCGA and for Casp3 (from OncoLnc)

We retrieve the clinical data directly using GDC_query. For the MK2 transcript data, this was downloaded from OncoLnc.org using the following parameters: Gene-MAPKAPK2, and percentile: 50/50. We did this because this allowed us to get transcript levels on all available patients (since upper and lower 50% would include all patients). We then calculated the quantiles manually, so that in the future, if we needed to change the threshold, we could do so without re-downloading the OncoLnc data. 

In this version, there is an additional step. Using TCGAbiolinks() we have manually extracted normalized RSEM reads from the LUAD dataset and compared it to what was given to us by OncoLnc


```{r}
library(knitr)
opts_chunk$set(tidy.opts=list(width.cutoff=50), tidy=TRUE)

```

```{r setup, include=TRUE}
knitr::opts_chunk$set(echo = TRUE)
rm(list=ls())
library("SummarizedExperiment")
library("dplyr")
library("DT")
library("TCGAbiolinks")
library(survival)
library(ggplot2)
library(ggpubr)
library(survminer)
library(knitr)
library("tidyr")
library("ggpmisc")

clinical <- GDCquery_clinic(project = "TCGA-LUAD", type = "clinical")
expr <- read.csv("../rawdata/LUAD.MAPKAPK2Exp.csv", header=TRUE)
colnames(expr) <- c("Patient", "MK2.Expression")
expr2 <- read.csv("../rawdata/LUAD.MAPKAPK2Exp.manual.csv", header=FALSE)
Casp3exp <- read.csv("../rawdata/LUAD.casp3Exp.2.csv")

colnames(expr2) <- c("MK2.Expression.Manual","Stage", "Patient")
expr2_normalcontrols <- expr2[grepl("11A", expr2$Patient),]
# take out normal controls
expr2 <- expr2[!grepl("11A", expr2$Patient),]

# remove non "01A" samples
expr3 <- expr2[grepl("01A", expr2$Patient),]
expr3$Patient <- substring(expr3$Patient, 1,12)
colnames(Casp3exp) <- c("Patient", "Casp3.Expression")
# the left_join. we need the clinical data, so we left_joined by TCGA patient id to the clinical dataset. This means that if there was a patient with clinical data without transcript levels, that would be a 'NA'. 
clinical <- left_join(clinical, expr, c("submitter_id"= "Patient"))
clinical <- left_join(clinical, expr3, c("submitter_id" = "Patient"))
clinical <- left_join(clinical, Casp3exp, c("submitter_id"="Patient"))

ggplot(clinical, aes(x=MK2.Expression,y=MK2.Expression.Manual))+geom_point()+theme_bw()+ylab("MAPKAPK2 transcript level - direct extract from LUAD")+xlab("MAPKAPK2 transcript level - OncoLnc")

clinical <- clinical[!is.na(clinical$vital_status),]
```


This shows that the correlation between MK2 Expression derived from OncoLnc and from TCGA is excellent. THus, we feel confident that the MK2 reads data from OncoLnc can be used going forward. This makes it easier to grab mRNA transcript levels for other genes directly from OncoLnc rather than having to extract it from TCGA via GDC.

## Clean the data, create a df for an LR and Cox PH analyses later

Now that we have left-joined the data, and checked to make sure that the join was done properly, we move on to prep the data for Cox PH work. 

```{r include=TRUE}

clinical$time <- mapply(function(x,y) if(is.na(x)) return(y) else return(x), clinical$days_to_death, clinical$days_to_last_follow_up) 
clinical$age_at_diagnosis <- clinical$age_at_diagnosis/365
clinical$race <- as.factor(clinical$race)
clinical$smoking <- lapply(clinical$cigarettes_per_day, function(x) if(!is.na(x)) return(1) else return(0))
clinical$tumor_grade <- as.factor(clinical$tumor_grade)
clinical$vital_status <- as.factor(clinical$vital_status)
clinical$status <- lapply(clinical$vital_status, function(x) if(x=="alive") return(0) else return(1))
clinical$smoking <- as.numeric(clinical$smoking)


# combine clinical stage

combineStage <- function(x) {
if(!is.na(x))  {
if(x=="Stage I" | x=="Stage IB" | x=="Stage IA") return("Early Stage")
if(x=="Stage II" | x=="Stage IIA" | x=="Stage IIB") return("Early Stage")
if(x=="Stage IIIA" | x=="Stage IIIB") return("Late Stage")
if(x=="Stage IV") return("Late Stage")
}
  } 

# combined histologic types 

combinePath <- function(x) {
 if(!is.na(x))  {
   
   if(x=="Adenocarcinoma, NOS") return("Adenocarcinoma")
   if(x=="Mucinous adenocarcinoma")  return("Adenocarcinoma")
   if(x=="Adenocarcinoma with mixed subtypes") return("Adenocarcinoma")
   if(x=="Papillary adenocarcinoma") return("Adenocarcinoma")
   else return("Non-Adenocarcinoma")
   
 }
  
}

# combine sub-stages into 1,2,3
clinical$ajcc_pathologic_stage_combined <- lapply(clinical$ajcc_pathologic_stage, combineStage)

clinical$ajcc_pathologic_stage_combined <- as.factor(as.character(clinical$ajcc_pathologic_stage_combined))

# combine histology types
clinical$primary_diagnosis_combined <- lapply(clinical$primary_diagnosis, combinePath)
clinical$primary_diagnosis_combined<- as.factor(unlist(clinical$primary_diagnosis_combined))

# get rid of NULL entries
clinical <- clinical[clinical$ajcc_pathologic_stage_combined!="NULL",]


# We already got rid of NULLs for ajcc_pathologic_stage. However, the factor levels still list 3 levels - early, late and NULL. To re-level, convert to character, re-convert to factor, and now only 2 levels. 
clinical$ajcc_pathologic_stage_combined<-as.factor(as.character(clinical$ajcc_pathologic_stage_combined))

clinical <- clinical[!is.na(clinical$MK2.Expression), ]

# create censor variable for CoxPH
clinical$censor <- lapply(clinical$vital_status, function(x) if(x=="Alive") return(0) else return(1))
clinical$censor <- as.numeric(clinical$censor)

# scale the MK2 variable
clinical$MK2.Expression <- clinical$MK2.Expression/1000
clinical$MK2.Expression.Manual <- clinical$MK2.Expression.Manual/1000
clinical$Casp3.Expression <- clinical$Casp3.Expression/1000
# create a simplified stage

combineStage.Simp <- function(x) {
if(!is.na(x))  {
if(x=="Stage I" | x=="Stage IB" | x=="Stage IA") return("Stage 1")
if(x=="Stage II" | x=="Stage IIA" | x=="Stage IIB") return("Stage 2")
if(x=="Stage IIIA" | x=="Stage IIIB") return("Stage 3")
if(x=="Stage IV") return("Stage 4")
}
  } 
clinical$ajcc_pathologic_stage_combined.2 <- lapply(clinical$ajcc_pathologic_stage, combineStage.Simp)
clinical$ajcc_pathologic_stage_combined.2 <- as.factor(unlist(clinical$ajcc_pathologic_stage_combined.2))

write.table(as.matrix(clinical), file="../rawdata/tcga-luad-clinical.processed.csv", row.names=FALSE, sep="\t", quote=FALSE, col.names = TRUE)

```


# Exploratory graphs

## Plot differences in transcript level by tumor histology and cancer stage

```{r}
# plot MK2 differences for adenocarcinoma only
ggplot(clinical[clinical$primary_diagnosis_combined=="Adenocarcinoma", ], aes(x=ajcc_pathologic_stage_combined, y=MK2.Expression, ))+geom_boxplot()+geom_point(position=position_jitter())+stat_compare_means()+theme_bw()+ylab("MK2 transcript level")


ggplot(clinical[clinical$primary_diagnosis_combined=="Adenocarcinoma", ], aes(x=ajcc_pathologic_stage_combined, y=MK2.Expression.Manual, ))+geom_boxplot()+geom_point(position=position_jitter())+stat_compare_means()+theme_bw()+ylab("MK2 transcript level")

```

This is a key difference: MK2 is clearly associated with tumor stage. Early stage tumors have more MK2 transcript than late stage tumors. Thus, going forward, we will stratify our analyses by Tumor stage. 

Here, we look at tumor differences more granularly (stage 1 vs. 2 vs. 3 vs. 4) and see a linear relationship - as stage increases, MK2 exp decreases. We also see that the LUAD cases that aren't really "AD" (non-Adenocarcinoma) don't share this relationship between Early/Late stage and MK2 transcript levels. So, when we prep the data for model work (below) we will restrict to Adenocarcinoma cases only

```{r}
ggplot(clinical[clinical$primary_diagnosis_combined=="Adenocarcinoma", ], aes(x=ajcc_pathologic_stage_combined.2, y=MK2.Expression, ))+geom_boxplot()+geom_point(position=position_jitter())+stat_compare_means()+theme_bw()+ylab("MK2 transcript level")

ggplot(clinical, aes(x=ajcc_pathologic_stage_combined, y=MK2.Expression, fill=primary_diagnosis_combined))+geom_boxplot()+geom_point(position=position_jitterdodge())+theme_bw()


```

## Explore the relationship between MK2 and Casp3

Here we look at the correlation between MK2 and Casp3 transcript levels. 

```{r}

ggplot(clinical, aes(x=MK2.Expression,y=Casp3.Expression)) + geom_point()+theme_bw()+geom_smooth(method="lm")
ggplot(clinical, aes(x=MK2.Expression.Manual,y=Casp3.Expression)) + geom_point()+theme_bw()

```


We are technically able to "fit" a line through this data, but the relationship (i.e slope) is not particularly impressive. The raw data does not suggest any sort of linear relationship between Casp3 mRNA and MK2 mRNA levels.

## Defining MK2 thresholds for "high" and "low" - Top 1/3 vs Bottom 2/3

Mechanistically, we're more interested in whether high vs. low MK2 transcript levels impact outcome. For this, we first categorize MK2 transcript levels based on quantiles and only take the lower and upper third in our groups for comparison. The MK2_Expression_topQ variable essentially captures this. If the transcript is in the top 33% then we return 1, everthing else (i.e the bottom 66%) returns 0. 

```{r}

# classify the MK2 expression variable as either top or bottom quantile, based on quantile definitions

bottom_quant <<- 0.33
top_quant <<- 0.66

clinical$MK2_Expression_bottomQ <- lapply(clinical$MK2.Expression, function (x) if(x < quantile(clinical$MK2.Expression, bottom_quant))  return(1) else return(0))

clinical$MK2_Expression_topQ <- lapply(clinical$MK2.Expression, function (x) if(x > quantile(clinical$MK2.Expression, top_quant))  return(1) else return(0))


```



## Changes in proportion of people who died across quantiles of MK2

### Prep data for using MK2 quantiles

The (poorly named) MK2_CoxPH_quantile df is the df we're using for all of our modeling purposes. This was originally used just for CoxPH, hence the name. We restrict this dataset to only those with follow-up available, and Adenocarcinomas only within LUAD. Additionally, we perform some more cleanup/create new variables as shown below.  

```{r}
# quantile and logistic regression

# this is the new data frame for this new round of calculations.  
MK2_CoxPH_quantile <- clinical[!is.na(clinical$time) & clinical$primary_diagnosis_combined=="Adenocarcinoma",]


# this provides the variable for R censoring at one year and the outcome variable for logit regression.
MK2_CoxPH_quantile$death_at_1year <- mapply(function(dead,time) if(dead==1 & time <366) return(1) else if(dead==0 & time > 366) return(0) else if (dead==0 & time < 366) return(0) else if (dead==1 & time > 366) return(0), MK2_CoxPH_quantile$censor, as.numeric(MK2_CoxPH_quantile$time))

MK2_CoxPH_quantile$death_at_2year <- mapply(function(dead,time) if(dead==1 & time <366*2) return(1) else if(dead==0 & time > 366*2) return(0) else if (dead==0 & time < 366*2) return(0) else if (dead==1 & time > 366*2) return(0), MK2_CoxPH_quantile$censor, as.numeric(MK2_CoxPH_quantile$time))

MK2_CoxPH_quantile$death_at_3year <- mapply(function(dead,time) if(dead==1 & time <366*3) return(1) else if(dead==0 & time > 366*3) return(0) else if (dead==0 & time < 366*3) return(0) else if (dead==1 & time > 366*3) return(0), MK2_CoxPH_quantile$censor, as.numeric(MK2_CoxPH_quantile$time))

# this piece of code above is critical - so let's double check and make sure everything got done right

ggplot(MK2_CoxPH_quantile, aes(x=submitter_id, y=time, color=vital_status))+ geom_point()+facet_wrap(~death_at_1year)+ylim(0,500)+xlab("Patient ID") + ylab("time to death")+geom_hline(yintercept = 365)


```


Interpretation off the plot above. So, we have colored each patient’s vital status by alive or dead, and we have faceted by whether they were assigned as “Death in one year” or not. So, if our code worked, all patients who died in less than one year (blue color; y < 365) should be classified as dead at one year (“1”). All patients who were alive in that time period should be classified as red and in the “0” column. After 365 days, everyone should have gotten a classification of “0” for death at one year, since they are all by definition still alive at 1 year, regardless of subsequent alive or dead status (color should mix, all colors in the “0” column). And that’s what we see…

```{r}
# lastly, going forward, we wish to show time in months rather than days, so we'll divide by 30
MK2_CoxPH_quantile$time <- MK2_CoxPH_quantile$time/30
```


### Proportion of deaths by MK2 quantiles

```{r}
#Now, moving on.

MK2_CoxPH_quantile$MK2_quantile <- ntile(MK2_CoxPH_quantile$MK2.Expression, 3)
MK2_CoxPH_quantile$Casp3_quantile  <- ntile(MK2_CoxPH_quantile$Casp3.Expression,3)
a <- aggregate(death_at_1year~MK2_quantile+ajcc_pathologic_stage_combined, FUN=sum, data=MK2_CoxPH_quantile)
b <- aggregate(death_at_1year~MK2_quantile+ajcc_pathologic_stage_combined, FUN=length, data=MK2_CoxPH_quantile)

quantile_df<-  as.data.frame(a$MK2_quantile)
quantile_df$stage <- as.data.frame(a$ajcc_pathologic_stage_combined)
quantile_df$prop <- as.data.frame(unlist(a$death_at_1year/b$death_at_1year))
quantile_df$prop <- unlist(quantile_df$prop)
colnames(quantile_df) <- c("quantile", "stage", "prop")
p1 <- ggplot(quantile_df[quantile_df$stage=="Early Stage",], aes(x=quantile, y=prop))+geom_point()+geom_smooth(method="lm", se=FALSE)+stat_poly_eq(formula="y~x", aes(label = paste(..eq.label.., ..rr.label.., sep = "~~~")), parse = TRUE)+theme_bw()+ylim(0,0.5)

p2 <- ggplot(quantile_df[quantile_df$stage=="Late Stage",], aes(x=quantile, y=prop))+geom_point()+geom_smooth(method="lm", se=FALSE)+stat_poly_eq(formula="y~x", aes(label = paste(..eq.label.., ..rr.label.., sep = "~~~")), parse = TRUE)+theme_bw()+ylim(0,0.5)

# This graph shows the relationship between MK2 quantile  and proportion of patients  who died in Early and Late Stage cancers. The relationship between MK2 quantile and Proportion of those patients  who died (denominator: all patients  in that quantile) 
gridExtra::grid.arrange(p1,p2, ncol=2)

```

# Logistic regression modeling death at 1 year ~ MK2+covariates

## Data clean-up
```{r}
MK2_CoxPH_quantile$death_at_1year <- as.factor(MK2_CoxPH_quantile$death_at_1year)
# logistic regression and ROC
MK2_CoxPH_quantile$ajcc_pathologic_stage_early_stage <- unlist(MK2_CoxPH_quantile$ajcc_pathologic_stage_early_stage)
MK2_CoxPH_quantile$MK2_quantile <- as.factor(MK2_CoxPH_quantile$MK2_quantile)
MK2_CoxPH_quantile$MK2_Expression_topQ <- as.factor(unlist(MK2_CoxPH_quantile$MK2_Expression_topQ))

```

## Logistic regression for death at 1 year, stratified by early/late, MK2 as a continuous variable

### Univariate regression

So, here we run a univariate analysis, with MK2 as a continuous variable, stratified by stage. We see a signal in early but not late stage disease. 

```{r}
# LR stratified by Early/Late
summary(glm(death_at_1year ~ MK2.Expression, data=MK2_CoxPH_quantile[MK2_CoxPH_quantile$ajcc_pathologic_stage_combined=="Early Stage", ], family=binomial(link="logit")))


summary(glm(death_at_1year ~ MK2.Expression, data=MK2_CoxPH_quantile[MK2_CoxPH_quantile$ajcc_pathologic_stage_combined=="Late Stage", ], family=binomial(link="logit")))


```


In this univariate regression, there is a suggestion of a negative association between MK2 (treated as a continuous variable) and death at one year, in the early stage patients. That is, higher MK2 = lower odds of death at one year. the CIs are imprecise, and thus the p values is 0.06. We do not see this in the late-stage patients. 


### Multivariate regression 

Now, we add covariates to the univariate model above. Again, because there is a baseline imbalance in transcript level between early and late stage, we are stratifying rather than adjusting. MK2 here remains a continuous variable, for now. 

```{r}
summary(glm(death_at_1year ~ MK2.Expression+smoking+age_at_diagnosis+gender, data=MK2_CoxPH_quantile[MK2_CoxPH_quantile$ajcc_pathologic_stage_combined=="Early Stage", ], family=binomial(link="logit")))

  summary(glm(death_at_1year ~ MK2.Expression+smoking+age_at_diagnosis+gender, data=MK2_CoxPH_quantile[MK2_CoxPH_quantile$ajcc_pathologic_stage_combined=="Late Stage", ], family=binomial(link="logit")))
```

Even with adjustment, there is again a suggestion of a negative association between MK2 transcript level and death at one year, with a p value of 0.1. There is a strong positive association between male gender and death at one year. 


## Model 2 in paper: Logistic regression for death at 1 year, stratified by early/late, but using Top 1/3 MK2 expression (dichotomous MK2 variable)

Now, we are interested in using MK2 as as dichotomous variable - high (top 1/3) or low (bottom 2/3). Clinically, with an appropriate reference cohort, this might be more translatable. Thus, we modeled death at one year, as a logistic regression, but now with MK2 as a dichotomous variable. The univariate analysis was significant, so here we run the multi-variate regression.

### Model 2 - Univariate Regression
```{r}
# LR stratified by Early/Late
summary(glm(death_at_1year ~ MK2_Expression_topQ, data=MK2_CoxPH_quantile[MK2_CoxPH_quantile$ajcc_pathologic_stage_combined=="Early Stage", ], family=binomial(link="logit")))


summary(glm(death_at_1year ~ MK2_Expression_topQ, data=MK2_CoxPH_quantile[MK2_CoxPH_quantile$ajcc_pathologic_stage_combined=="Late Stage", ], family=binomial(link="logit")))
```

We see highly significant association between high MK2 expression (as a dichotomous variable) and death at 1 year, in early but not late stage cancer. 


### Model 2: Multivariate regression 

So, now we will run 2 models. MK2 as dichotomous variable, with covariates, stratified by stage with the outcome of death at 1 year, as well as death at 2 years. 

Here we first look at Early Stage only

```{r}
logreg_topq1_early <- glm(death_at_1year ~ MK2_Expression_topQ+smoking+age_at_diagnosis+gender, data=MK2_CoxPH_quantile[MK2_CoxPH_quantile$ajcc_pathologic_stage_combined=="Early Stage", ], family=binomial(link="logit"))


summary(logreg_topq1_early)
exp(logreg_topq1_early$coefficients)
exp(confint.lm(logreg_topq1_early))

logreg_topq1_early_2y <- glm(death_at_2year ~ MK2_Expression_topQ+smoking+age_at_diagnosis+gender, data=MK2_CoxPH_quantile[MK2_CoxPH_quantile$ajcc_pathologic_stage_combined=="Early Stage", ], family=binomial(link="logit"))


summary(logreg_topq1_early_2y)
exp(logreg_topq1_early_2y$coefficients)
exp(confint.lm(logreg_topq1_early_2y))
```

Now we look at late stage

```{r}
logreg_topq1_late <- glm(death_at_1year ~ MK2_Expression_topQ+smoking+age_at_diagnosis+gender, data=MK2_CoxPH_quantile[MK2_CoxPH_quantile$ajcc_pathologic_stage_combined=="Late Stage", ], family=binomial(link="logit"))

summary(logreg_topq1_late)
exp(logreg_topq1_late$coefficients)
exp(confint.lm(logreg_topq1_late))

logreg_topq1_late_2y <- glm(death_at_2year ~ MK2_Expression_topQ+smoking+age_at_diagnosis+gender, data=MK2_CoxPH_quantile[MK2_CoxPH_quantile$ajcc_pathologic_stage_combined=="Late Stage", ], family=binomial(link="logit"))


summary(logreg_topq1_late_2y)
exp(logreg_topq1_late_2y$coefficients)
exp(confint.lm(logreg_topq1_late_2y))
```


### Model 2: MK2 multivariate model with Casp3

Now we run the multi-variate model again but with casp3 expression
```{r}

summary(glm(death_at_2year ~ MK2_Expression_topQ+smoking+age_at_diagnosis+gender+Casp3.Expression, data=MK2_CoxPH_quantile[MK2_CoxPH_quantile$ajcc_pathologic_stage_combined=="Early Stage", ], family=binomial(link="logit")))

```

Addition of Casp3 does not change the effect of MK2_expression on death at 2 year in this model. 


# Cox Proportional Hazards modeling, with right censoring at 1 year. 

## Data Cleaning

```{r}
# data cleaning prior to Cox PH model
MK2_CoxPH_quantile$MK2_Expression_topQ <- as.factor(unlist(MK2_CoxPH_quantile$MK2_Expression_topQ))
MK2_CoxPH_quantile$gender <- as.factor(MK2_CoxPH_quantile$gender)

# be careful here. as.numeric alone will return the factor level, not the value!
MK2_CoxPH_quantile$censor <- as.numeric(as.character(MK2_CoxPH_quantile$death_at_1year))
MK2_CoxPH_quantile$censor.2y <- as.numeric(as.character(MK2_CoxPH_quantile$death_at_2year))
MK2_CoxPH_quantile$censor.3y <- as.numeric(as.character(MK2_CoxPH_quantile$death_at_3year))

# there should be no discrepancy - only two dots should show up
plot(MK2_CoxPH_quantile$censor, MK2_CoxPH_quantile$death_at_1year)


# write the final version for use in HR bootstrapping analyses
write.table(as.matrix(MK2_CoxPH_quantile), file="../rawdata/tcga-luad-clinical.processed_forCoxPH.csv", row.names=FALSE, sep="\t", quote=FALSE, col.names = TRUE)



```

## Cox PH model R censored at one year, stratified by early/late, MK2 as a continuous variable

### Model 1: MK2 only
```{r}
# univariate Cox PH
summary(coxph(Surv(time, censor) ~ MK2.Expression, id=submitter_id, data=MK2_CoxPH_quantile[MK2_CoxPH_quantile$ajcc_pathologic_stage_combined=="Early Stage",]))

# Multi-variate Cox PH 
MK2_coxph_model_tb_1y_early <- coxph(Surv(time, censor) ~ MK2.Expression+gender+smoking+age_at_diagnosis, id=submitter_id, data=MK2_CoxPH_quantile[MK2_CoxPH_quantile$ajcc_pathologic_stage_combined=="Early Stage",])

MK2_coxph_model_tb_1y_late <- coxph(Surv(time, censor) ~ MK2.Expression+gender+smoking+age_at_diagnosis, id=submitter_id, data=MK2_CoxPH_quantile[MK2_CoxPH_quantile$ajcc_pathologic_stage_combined=="Late Stage",])


# PH testing
cox.zph(MK2_coxph_model_tb_1y_early)
cox.zph(MK2_coxph_model_tb_1y_late)

# results
summary(MK2_coxph_model_tb_1y_early)
summary(MK2_coxph_model_tb_1y_late)
```

So here, we can see that in multi-variate Cox PH model, we see a signal for MK2 and improved survival, but only in the early stage tumors. 


### MK2 with casp 3

```{r}
summary(coxph(Surv(time, censor) ~ MK2.Expression+gender+smoking+age_at_diagnosis+Casp3.Expression, id=submitter_id, data=MK2_CoxPH_quantile[MK2_CoxPH_quantile$ajcc_pathologic_stage_combined=="Early Stage",]))

```
Similar to our logistic reg model, no real difference with Casp3 addition to the model. 

## Model 1 in paper: Cox PH model R censored at one year and two year, stratified by early late, MK2 as dichotomous (top 1/3 vs. bottom 2/3)

### Model 1: MK2+other covariates, at 1 year and 2 year 
```{r}

summary(coxph(Surv(time, censor) ~ MK2_Expression_topQ+gender+smoking+age_at_diagnosis, id=submitter_id, data=MK2_CoxPH_quantile[MK2_CoxPH_quantile$ajcc_pathologic_stage_combined=="Early Stage",]))

summary(coxph(Surv(time, censor) ~ MK2_Expression_topQ+gender+smoking+age_at_diagnosis, id=submitter_id, data=MK2_CoxPH_quantile[MK2_CoxPH_quantile$ajcc_pathologic_stage_combined=="Late Stage",]))

summary(coxph(Surv(time, censor.2y) ~ MK2_Expression_topQ+gender+smoking+age_at_diagnosis, id=submitter_id, data=MK2_CoxPH_quantile[MK2_CoxPH_quantile$ajcc_pathologic_stage_combined=="Early Stage",]))

summary(coxph(Surv(time, censor.2y) ~ MK2_Expression_topQ+gender+smoking+age_at_diagnosis, id=submitter_id, data=MK2_CoxPH_quantile[MK2_CoxPH_quantile$ajcc_pathologic_stage_combined=="Late Stage",]))

```

### Model 1: MK2 + other covariates + casp3, at 1 year and 2 year
```{r}
summary(coxph(Surv(time, censor) ~ MK2_Expression_topQ+gender+smoking+age_at_diagnosis+Casp3.Expression, id=submitter_id, data=MK2_CoxPH_quantile[MK2_CoxPH_quantile$ajcc_pathologic_stage_combined=="Early Stage",]))
```

So, the same pattern here - we don't really see an impact of adding casp3 to the multivariable Cox PH model. 

# Model Graphics

## KM curve - here, we're going to stratify MK2 by "hi" vs. "low" - Model 1

```{r}
library(survminer)
library(survival)
library(ggsci)
km_early <- survfit(Surv(time, censor)~MK2_Expression_topQ, data=MK2_CoxPH_quantile[MK2_CoxPH_quantile$ajcc_pathologic_stage_combined=="Early Stage",], type='kaplan-meier')
ggsurvplot(km_early, xlim=c(0,12), conf.int = TRUE, break.x.by=2, pval=TRUE, font.y=14,font.x=14,font.tickslab=14, ggtheme=theme_bw())

km_late <- survfit(Surv(time, censor)~MK2_Expression_topQ, data=MK2_CoxPH_quantile[MK2_CoxPH_quantile$ajcc_pathologic_stage_combined=="Late Stage",], type='kaplan-meier')
ggsurvplot(km_late, xlim=c(0,12), conf.int = TRUE, break.x.by=2, pval=TRUE, font.y=14,font.x=14,font.tickslab=14, ggtheme=theme_bw())


km_early.2y <- survfit(Surv(time, censor.2y)~MK2_Expression_topQ, data=MK2_CoxPH_quantile[MK2_CoxPH_quantile$ajcc_pathologic_stage_combined=="Early Stage",], type='kaplan-meier')
ggsurvplot(km_early.2y, xlim=c(0,24), conf.int = TRUE, break.x.by=2, pval=TRUE, font.y=14,font.x=14,font.tickslab=14, ggtheme=theme_bw())

km_late.2y <- survfit(Surv(time, censor.2y)~MK2_Expression_topQ, data=MK2_CoxPH_quantile[MK2_CoxPH_quantile$ajcc_pathologic_stage_combined=="Late Stage",], type='kaplan-meier')
ggsurvplot(km_late.2y, xlim=c(0,24), conf.int = TRUE, break.x.by=2, pval=TRUE, font.y=14,font.x=14,font.tickslab=14, ggtheme=theme_bw())



```

## Predicted probabilities graphs for Logistic Regression (Model 2)
```{r}
library(effects)
library(ggthemes)
MK2efx_early <- Effect("MK2_Expression_topQ", logreg_topq1_early_2y)
MK2efx_late <- Effect("MK2_Expression_topQ", logreg_topq1_late_2y)

MK2efx_early <- as.data.frame(MK2efx_early)
MK2efx_late <- as.data.frame(MK2efx_late)
MK2efx_early$stage <- "Early"
MK2efx_late$stage <- "Late"
MK2efx <- rbind(MK2efx_early, MK2efx_late)
MK2efx$MK2_Expression_topQ <- lapply(MK2efx$MK2_Expression_topQ, function(x) if (x==0) return("Low") else return("High"))
MK2efx$MK2_Expression_topQ <- as.factor(unlist(MK2efx$MK2_Expression_topQ))
ggplot(MK2efx, aes(x=stage, y=fit, group=MK2_Expression_topQ, color=MK2_Expression_topQ, ymin=lower,ymax=upper))+geom_point(position=position_dodge(width=0.2))+geom_errorbar(width=0, position=position_dodge(width=0.2))+theme_bw(base_size = 14)+ylab("Predicted  Probability of death at 2 years")+coord_flip()


```
# Demographics / Table 1 
```{r}

library(tableone)
listVars <- c("age_at_diagnosis", "gender", "smoking", "ajcc_pathologic_stage","ajcc_pathologic_n", "prior_malignancy", "primary_diagnosis", "time", "race", "MK2.Expression")
catVars <- c("gender", "smoking", "ajcc_pathologic_stage", "ajcc_pathologic_n", "prior_malignancy", "primary_diagnosis", "race") 

# Table 1 where we stratify by Early/Late Stage
table1 <- CreateTableOne(listVars, clinical, catVars, strata=c("ajcc_pathologic_stage_combined"), addOverall=TRUE)
clinical$MK2_Expression_topQ_t1 <- unlist(clinical$MK2_Expression_topQ)
kable(print(table1))
# Table 1 where we stratify by MK2 "hi" vs. "low"
table1b <- CreateTableOne(listVars, clinical, catVars, strata=c("MK2_Expression_topQ_t1"), addOverall=TRUE)
kable(print(table1b))
```


# Additional models and analysis

## Continuous MK2 expression, no R censoring:  Survival ~ expression (continuous) + clinical stage (stage 1, stage 2,...)

In this model, we use MK2 expression as a continuous variable - in other words, don't split into low vs. high (for that, see below). The 'MK.Expression' variable is the MK2 transcript level. Here, we're going to use the full follow-up time without any R censoring.

```{r}


# COX PH model
coxph_model_all <- coxph(Surv(time, censor.2y) ~ gender+smoking+MK2.Expression, id=submitter_id, data=MK2_CoxPH_quantile[MK2_CoxPH_quantile$ajcc_pathologic_stage_combined=="Early Stage",])


# examine MK2 expression values 
hist(clinical$MK2.Expression)
cox.zph(coxph_model_all)
plot(cox.zph(coxph_model_all)[3])
summary(coxph_model_all)

```


## Logistic regression for death at 1 year, stratified by specific stage

As shown above, LR using MK2 as a continuous variable but tumor stage dichotomized as early/late produced some interesting results. We wondered if we were "overlumping" stage by combining Stage I and II into "Early Stage". Here we split stages into 1-4. Of course, now the problem was low numbers. These models did not yield any additional information, so were not considered further. 

```{r}
# LR stratified by individual stage
summary(glm(death_at_1year ~ MK2.Expression, data=MK2_CoxPH_quantile[MK2_CoxPH_quantile$ajcc_pathologic_stage_combined.2=="Stage 1", ], family=binomial(link="logit")))
summary(glm(death_at_1year ~ MK2.Expression, data=MK2_CoxPH_quantile[MK2_CoxPH_quantile$ajcc_pathologic_stage_combined.2=="Stage 2", ], family=binomial(link="logit")))
summary(glm(death_at_1year ~ MK2.Expression, data=MK2_CoxPH_quantile[MK2_CoxPH_quantile$ajcc_pathologic_stage_combined.2=="Stage 3", ], family=binomial(link="logit")))
summary(glm(death_at_1year ~ MK2.Expression, data=MK2_CoxPH_quantile[MK2_CoxPH_quantile$ajcc_pathologic_stage_combined.2=="Stage 4", ], family=binomial(link="logit")))

summary(glm(death_at_1year ~ MK2_Expression_topQ, data=MK2_CoxPH_quantile[MK2_CoxPH_quantile$ajcc_pathologic_stage_combined.2=="Stage 1", ], family=binomial(link="logit")))
summary(glm(death_at_1year ~ MK2_Expression_topQ, data=MK2_CoxPH_quantile[MK2_CoxPH_quantile$ajcc_pathologic_stage_combined.2=="Stage 2", ], family=binomial(link="logit")))
summary(glm(death_at_1year ~ MK2_Expression_topQ, data=MK2_CoxPH_quantile[MK2_CoxPH_quantile$ajcc_pathologic_stage_combined.2=="Stage 3", ], family=binomial(link="logit")))
summary(glm(death_at_1year ~ MK2_Expression_topQ, data=MK2_CoxPH_quantile[MK2_CoxPH_quantile$ajcc_pathologic_stage_combined.2=="Stage 4", ], family=binomial(link="logit")))

summary(glm(death_at_1year ~ MK2_Expression_topQ, data=MK2_CoxPH_quantile[MK2_CoxPH_quantile$ajcc_pathologic_stage_combined.2=="Stage 1", ], family=binomial(link="logit")))
summary(glm(death_at_1year ~ MK2_Expression_topQ, data=MK2_CoxPH_quantile[MK2_CoxPH_quantile$ajcc_pathologic_stage_combined.2=="Stage 2", ], family=binomial(link="logit")))
summary(glm(death_at_1year ~ MK2_Expression_topQ, data=MK2_CoxPH_quantile[MK2_CoxPH_quantile$ajcc_pathologic_stage_combined.2=="Stage 3", ], family=binomial(link="logit")))
summary(glm(death_at_1year ~ MK2_Expression_topQ, data=MK2_CoxPH_quantile[MK2_CoxPH_quantile$ajcc_pathologic_stage_combined.2=="Stage 4", ], family=binomial(link="logit")))
```
## Jackknife-type analysis for effects due to long period of patient recruitment

It appears that patients were recruited across many years. Chemo regimens change with time. To Figure out whether this plays a role, we re-estimate the odds ratio for high MK2 levels in the multivariate Linear Regression Model (Model 2)  multiple times, each time removing only the patients recruited in a particular year

```{r}
MK2_tvary <- MK2_CoxPH_quantile
years <- as.data.frame(table(as.factor(as.character(clinical$year_of_diagnosis))))
MK2_tvary$year_of_diagnosis <- as.factor(as.character(MK2_tvary$year_of_diagnosis))


returnPointEst <- function(x) {
   modelresult <- glm(death_at_1year ~ MK2_Expression_topQ+smoking+age_at_diagnosis+gender, data=MK2_CoxPH_quantile[MK2_CoxPH_quantile$ajcc_pathologic_stage_combined=="Early Stage", ], family=binomial(link="logit"))
   return(exp(modelresult$coefficients[1]))
}

years$estimate <- lapply(as.character(years$Var1), returnPointEst)
years$estimate <- lapply(years$estimate, function(x) return(x[[1]]))

returnPointEstlci <- function(x) {
    modelresult <- glm(death_at_1year ~ MK2_Expression_topQ+smoking+age_at_diagnosis+gender, data=MK2_CoxPH_quantile[MK2_CoxPH_quantile$ajcc_pathologic_stage_combined=="Early Stage", ], family=binomial(link="logit"))
   return(exp(confint.lm(modelresult)[2]))
}  

returnPointEstuci <- function(x) {
     modelresult <- glm(death_at_1year ~ MK2_Expression_topQ+smoking+age_at_diagnosis+gender, data=MK2_CoxPH_quantile[MK2_CoxPH_quantile$ajcc_pathologic_stage_combined=="Early Stage", ], family=binomial(link="logit"))
   return(exp(confint.lm(modelresult)[7]))
}  


years$estimate_lci <- lapply(as.character(years$Var1), returnPointEstlci) 
years$estimate_uci <- lapply(as.character(years$Var1), returnPointEstuci) 

years$estimate <- as.numeric(years$estimate)
years$estimate_lci<- as.numeric(years$estimate_lci)
years$estimate_uci <- as.numeric(years$estimate_uci)

ggplot(years, aes(x=Var1, y=estimate))+geom_point()+geom_errorbar(aes(ymin=estimate_lci, ymax=estimate_uci), width=0)+theme_bw()+ylim(0,1.2)+geom_hline(yintercept = 1)+ylab("OR for death at one year as a function of High MK2 level")+xlab("Patient Recruitment Year that was excluded from model")+theme(axis.text.x = element_text(angle = 90, hjust = 1))

```
# Normal vs. tumor tissue

One of the benefits of downloading data from the TCGA is that it includes the "11" barcode, which is normal tissue
```{r}
expr2_normalcontrols$Patient <- substring(expr2_normalcontrols$Patient,1,12)
expr2_normalcontrols$sample.type="Normal Tissue"
expr2_t <- expr3
expr2_t$sample.type="Tumor"

exp2_paired <- left_join(expr2_normalcontrols, expr2_t, by=c("Patient"))
exp2_paired <-exp2_paired[!is.na(exp2_paired$MK2.Expression.Manual.y),]
colnames(exp2_paired) <- c("MK2.Expression.Normal", "Stage.1" , "Patient","sample.type.1", "MK2.Expression.Tumor", "Stage.2", "sample.type.2")

# These data are plotted elsewhere. (See graphpad file)

# look at FC in MK2 as a predictor of survival

exp2_paired$MK2_diff <- exp2_paired$MK2.Expression.Tumor-exp2_paired$MK2.Expression.Normal
exp2_paired$MK2_FC <- exp2_paired$MK2.Expression.Tumor/exp2_paired$MK2.Expression.Normal
exp2_paired$Patient <- substring(exp2_paired$Patient, 1,12)

exp2_paired <- left_join(exp2_paired, clinical, by=c("Patient"="submitter_id"))

ggplot(exp2_paired, aes(x=Stage.1, y=MK2_diff, fill=vital_status))+geom_boxplot()+stat_compare_means()
ggplot(exp2_paired, aes(x=Stage.1, y=MK2.Expression.Normal, fill=vital_status))+geom_boxplot()+stat_compare_means()
ggplot(exp2_paired, aes(x=Stage.1, y=MK2.Expression.Tumor, fill=vital_status))+geom_boxplot()+stat_compare_means()

summary(coxph(Surv(time, censor) ~ MK2_FC, id=Patient, data=exp2_paired))



```


# Hsp27 and MK2

Here, we explore correlations between Hsp27 and MK2
```{r}

HSP27Exp <- read.csv("../rawdata/LUAD.Hsp27Exp.csv")

HSP27Exp <- left_join(HSP27Exp, expr, by=c("Patient"))
library(ggpubr)
ggplot(HSP27Exp, aes(x=log(Expression), y=log(MK2.Expression)))+geom_point()+geom_smooth(method="lm")+theme_bw()+stat_cor()

```
# System Info
```{r}
sessionInfo()
```